users/tests/test_models.py

```diff
@@ -255,3 +261,56 @@ def test_user_serialize(snapshot, project):
     assert child_without_enrolments.ticket_system_passwords.count() == 0
     user.refresh_from_db()
     snapshot.assert_match(user.serialize())
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "notification_type,needs_acceptance",
+    [
+        (EventNotificationType.EVENT_PUBLISHED, True),
+        (EventNotificationType.EVENT_GROUP_PUBLISHED, True),
+        (EventNotificationType.OCCURRENCE_ENROLMENT, False),
+        (EventNotificationType.OCCURRENCE_UNENROLMENT, False),
+        (EventNotificationType.OCCURRENCE_CANCELLED, False),
+        (EventNotificationType.OCCURRENCE_REMINDER, False),
+        (EventNotificationType.OCCURRENCE_FEEDBACK, False),
+        (ChildrenNotificationType.SIGNUP, False),
+        (UserNotificationType.GUARDIAN_EMAIL_CHANGE_TOKEN, False),
+        (UserNotificationType.GUARDIAN_EMAIL_CHANGED, False),
+        (SubscriptionsNotificationType.FREE_SPOT, False),
+    ],
+)
+def test_has_accepted_communication_for_notification(
+    notification_type, needs_acceptance
+):
+    """Some of the notification types needs an acceptance for communication.
+    Rest are so called transactional notifications which cannot be rejected.
+    Test that the GuardianQuerySet.has_accepted_communication_for_notification,
+    checks the acceptance for the types that needs it.
+
+    Args:
+        notification_type (str): notification type
+        needs_acceptance (bool): True if the notitication type needs an acceptance
+    """
+    guardian_with_filtered_communication = GuardianFactory(
+        has_accepted_communication=False
+    )
+    guardian_with_all_communication = GuardianFactory(has_accepted_communication=True)
+
+    queryset_result = list(
+        Guardian.objects.has_accepted_communication_for_notification(notification_type)
+    )
+
+    if needs_acceptance:
+        assert len(queryset_result) == 1
+        assert guardian_with_all_communication in queryset_result
+        assert guardian_with_filtered_communication not in queryset_result
+    else:
+        assert len(queryset_result) == 2
+        assert all(
+            guardian in queryset_result
+            for guardian in [
+                guardian_with_all_communication,
+                guardian_with_filtered_communication,
+            ]
+        )
```

Comment:
How would you feel about splitting this into two test cases and making them look a bit more alike in their testing? If you find this valuable you can revamp it as you like:
```suggestion
@pytest.mark.django_db
@pytest.mark.parametrize(
    "notification_type",
    [
        EventNotificationType.OCCURRENCE_ENROLMENT,
        EventNotificationType.OCCURRENCE_UNENROLMENT,
        EventNotificationType.OCCURRENCE_CANCELLED,
        EventNotificationType.OCCURRENCE_REMINDER,
        EventNotificationType.OCCURRENCE_FEEDBACK,
        ChildrenNotificationType.SIGNUP,
        UserNotificationType.GUARDIAN_EMAIL_CHANGE_TOKEN,
        UserNotificationType.GUARDIAN_EMAIL_CHANGED,
        SubscriptionsNotificationType.FREE_SPOT,
    ],
)
def test_has_accepted_communication_for_notification_irrefusables(
    notification_type
):
    """Some of the notification types do not need an acceptance for communication.
    They are so called transactional notifications which cannot be rejected.
    Test that the GuardianQuerySet.has_accepted_communication_for_notification,
    does not check the acceptance for the types that don't need it.

    Args:
        notification_type (str): notification type
    """
    guardian_with_filtered_communication = GuardianFactory(
        has_accepted_communication=False
    )
    guardian_with_all_communication = GuardianFactory(has_accepted_communication=True)

    queryset_result = list(
        Guardian.objects.has_accepted_communication_for_notification(notification_type)
    )
    
    assert len(queryset_result) == 2
    assert guardian_with_all_communication in queryset_result
    assert guardian_with_filtered_communication in queryset_result



@pytest.mark.django_db
@pytest.mark.parametrize(
    "notification_type",
    [
        EventNotificationType.EVENT_PUBLISHED,
        EventNotificationType.EVENT_GROUP_PUBLISHED,
    ],
)
def test_has_accepted_communication_for_notification_needs_acceptance(
    notification_type
):
    """Some of the notification types need an acceptance for communication.
    Test that the GuardianQuerySet.has_accepted_communication_for_notification,
    checks the acceptance for the types that need it.

    Args:
        notification_type (str): notification type
    """
    guardian_with_filtered_communication = GuardianFactory(
        has_accepted_communication=False
    )
    guardian_with_all_communication = GuardianFactory(has_accepted_communication=True)

    queryset_result = list(
        Guardian.objects.has_accepted_communication_for_notification(notification_type)
    )

    assert len(queryset_result) == 1
    assert guardian_with_all_communication in queryset_result
    assert guardian_with_filtered_communication not in queryset_result
```